rotkehlchen/tests/unit/test_general_cache.py

```diff
@@ -110,8 +147,42 @@ def mock_call_contract(contract, node_inquirer, method_name, **kwargs):
         new=mock_call_contract,
     )
 
+    requests_get = requests.get
+
+    def mock_requests_get(url, timeout):  # pylint: disable=unused-argument
+        """Mock requests.get.
+        requests.get in this test should be used only for curve api queries.
+        If use_curve_api is False, return success: False so that curve cache query fallbacks to
+        chain query.
+        If use_curve_api is True, then to minimize number of pools check in this test, return
+        empty pools list for 3 out of the 4 curve api endpoints and for the remaining one perform
+        a real request and take from the response only 2 first pools.
+        """
+        if url not in CURVE_API_URLS:
+            raise AssertionError(f'Unexpected url {url} was called')
+
+        if use_curve_api is False:
+            return MockResponse(status_code=200, text=json.dumps({'success': False, 'data': {'poolData': []}}))  # noqa: E501
+
+        if url != CURVE_API_URLS[0]:
+            return MockResponse(status_code=200, text=json.dumps({'success': True, 'data': {'poolData': []}}))  # noqa: E501
+
+        response_json = requests_get(url).json()
+        response_json['data']['poolData'] = response_json['data']['poolData'][:2]
+        return MockResponse(status_code=200, text=json.dumps(response_json))
+
+    requests_patch = patch(
+        'requests.get',
+        side_effect=mock_requests_get,
+    )
+
+    # Since DAI (which is a constant asset) was deleted at the beginning of this test and DAI is
+    # used in one of the tested curve pools, we disable constant assets here (otherwise the code
+    # will attempt to load DAI from packaged db, won't find it there and will raise an error).
+    constant_assets_patch = patch('rotkehlchen.constants.assets.CONSTANT_ASSETS', [])
```

Comment:
This seems like a hack complicating the test. Please if you want to delete an asset delete any non-constant asset.